### PR TITLE
Consolidate TEAL and AVM versioning

### PR DESCRIPTION
**Pull request by jdtzmn**
This PR replaces `TEAL version` language with `program version`. In certain places where the language made more sense, `AVM` was used instead of `program`.

Some considerations:
* Language in `CHANGELOG.md` was left unchanged.
* `verifyTealVersion` was renamed to `verifyProgramVersion`. My assumption is that no deprecation notice is needed as `verifyTealVersion` is not listed in the `__init__.py`.
* The following breaking renames were made. The `*_TEAL_VERSION` constants are still exported for the time being, but have been marked as deprecated.
    * `MAX_TEAL_VERSION` → `MAX_PROGRAM_VERSION`
    * `MIN_TEAL_VERSION` → `MIN_PROGRAM_VERSION`
    * `DEFAULT_TEAL_VERSION` → `DEFAULT_PROGRAM_VERSION`

### CODE REVIEW

File: docs/arithmetic_expression.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ The associativity and precedence of the overloaded Python arithmetic operators a
 Byteslice Arithmetic
 --------------------
 
-Byteslice arithemetic is available for Teal V4 and above. 
+Byteslice arithemetic is available for AVM V4 and above.
```
Comment (michaeldiamant) [inline docs/arithmetic_expression.rst:52]: @jdtzmn Minor - Noticed unrelated typo while reviewing.

```suggestion
Byteslice arithmetic is available for AVM V4 and above. 
```

File: docs/versions.rst [hunk 2]
```diff
@@ -1,14 +1,14 @@
 .. _versions:
 
-TEAL Versions
+AVM Versions
 =============
 
-Each version of PyTeal compiles contracts for a specific version of TEAL. Newer versions of TEAL
+Each version of PyTeal compiles contracts for a specific version of AVM. Newer versions of the AVM
```
Comment (michaeldiamant) [inline docs/versions.rst:6]: @jdtzmn Minor:  Propose either the _suggestion_ or "_the_ AVM".
```suggestion
Each version of PyTeal compiles contracts for a specific AVM version. Newer versions of the AVM
```

File: docs/versions.rst [hunk 3]
```diff
@@ -18,14 +18,14 @@ TEAL Version PyTeal Version
 6            >= 0.10.0
```
Comment (michaeldiamant) [inline docs/versions.rst:18]: @jdtzmn 2 requests here relating to our desire to have the change take effect as of AVM 7:
* Can you extend the table to include AVM 7 with expected version?  As of writing, I think it'll be v0.14.0.
* Can you base the PR off of `teal7` (https://github.com/algorand/pyteal/pull/416)?
  * I realize the request creates conflicts.  I apologize for not calling it out earlier.

File: pyteal/ast/opup.py [hunk 4]
```diff
@@ -29,14 +29,14 @@ class OpUpMode(Enum):
     OnCall = 1
 
 
-ON_CALL_APP = Bytes("base16", "068101")  # v6 teal program "int 1"
+ON_CALL_APP = Bytes("base16", "068101")  # v6 avm program "int 1"
```
Comment (jasonpaulos) [inline pyteal/ast/opup.py:32]: We should probably retain "teal" here, since it describes the opcode language

**Review by jasonpaulos — COMMENTED** [review]
Looks good to me, I just have one comment

File: pyteal/ast/opup.py (hunk 4 above)
Comment (michaeldiamant) [reply pyteal/ast/opup.py:32]: > Our terminology around that isn't changing, is it?

No - terminology not changing here.  Another alternative is to write `v6 program`.

**Review by michaeldiamant — APPROVED** [review]
@jdtzmn Thanks for the effort here!  ☕ 

Leaving approval under the assumption that open threads are resolved prior to merge, and https://github.com/algorand/pyteal/pull/450 is merged into `teal7`, and then merged into this PR.

**Comment by jasonpaulos** [comment]
Be aware that `InnerTxnBuilder.Execute` was just merged into `master`, which references "TEAL version" in its docstring

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**